From: Rik van Riel <riel@redhat.com>
Date: Mon, 27 Aug 2007 17:55:22 -0400
Subject: [xen] export cpu_llc_id as gpl
Message-id: 46D3484A.3080909@redhat.com
O-Subject: [PATCH RHEL5.1] EXPORT_SYMBOL_GPL crusade
Bugzilla: 429004

Of those 4, flush_tlb_current_task, flush_tlb_mm and raw_safe_halt
are EXPORT_SYMBOLed in Xen but not exported at all on non-Xen.
However, these three are all called from inline functions in
include files, which makes me a little nervous about removing
them.

The last symbol, cpu_llc_id, is not referenced from any kernel
modules. at all.  That means that one is safe to change.

IMHO this EXPORT_SYMBOL(_GPL) thing is probably not important
enough to risk regressions, so I am trying to play it as safely
as possible from the regression point of view.

--
All Rights Reversed

diff --git a/drivers/xen/core/smpboot.c b/drivers/xen/core/smpboot.c
index 938c61a..2a4513c 100644
--- a/drivers/xen/core/smpboot.c
+++ b/drivers/xen/core/smpboot.c
@@ -44,7 +44,7 @@ int cpu_llc_id[NR_CPUS] __cpuinitdata = {[0 ... NR_CPUS-1] = BAD_APICID};
 #elif defined(__x86_64__)
 u8 cpu_llc_id[NR_CPUS] __cpuinitdata  = {[0 ... NR_CPUS-1] = BAD_APICID};
 #endif
-EXPORT_SYMBOL(cpu_llc_id);
+EXPORT_SYMBOL_GPL(cpu_llc_id);
 
 cpumask_t cpu_online_map;
 EXPORT_SYMBOL(cpu_online_map);
